doc/server/errors.md

```diff
@@ -1,14 +1,24 @@
 # Error handling
 
-## Exception handling
+Error handling in tapir is divided into three areas:
 
-There's no exception handling built into tapir. However, tapir contains a more general error handling mechanism, as the
-endpoints can contain dedicated error outputs.
+1. Error outputs: defined per-endpoint, used for errors handled by the business logic
+2. Failed effects: exceptions which are not handled by the business logic (corresponds to 5xx responses)    
+3. Decode failures: format errors, when the input values can't be decoded (corresponds to 4xx responses, or 
+   trying another endpoint)
 
-If the logic function, which is passed to the server interpreter, fails (i.e. throws an exception, which results in
-a failed `Future` or `IO`/`Task`), this is propagated to the library (akka-http or http4s). 
+While 1. is specific to an endpoint, handlers for 2. and 3. are typically the same for multiple endpoints, and are
+specified as part of the server's interpreter [options](options.md).
+
+## Error outputs
+
+Each endpoint can contain dedicated error outputs, in addition to outputs which are used in case of success. The 
+business logic can then return either an error value, or a success value. Any business-logic-level errors should
+be signalled this way. This can include validation, failure of downstream services, or inability to server the request
```

Comment:
think It's a typo here -> _inability to server_

Reply:
Thx :)